chef/cookbooks/provisioner/templates/suse/crowbar_register.erb

```diff
@@ -359,8 +365,16 @@ fi
 # Steps from sledgehammer
 # -----------------------
 
+MAC=$(cat /sys/class/net/$BOOTDEV/address)
 if test $KEEP_EXISTING_HOSTNAME -ne 1; then
+  HOSTNAME="d${MAC//:/-}.${DOMAIN}"
   echo "$HOSTNAME" > /etc/HOSTNAME
+else
+  HOSTNAME=$(cat /etc/HOSTNAME)
+  if ! echo "$HOSTNAME" | grep -q -o "\.$DOMAIN$" || echo "${HOSTNAME%%.$DOMAIN}" | grep -q -o "\."; then
+    echo "The fully qualified domain name did not contain the $DOMAIN cloud domain."
+    exit 1
+  fi
```

Comment:
It's not removed, it's at https://github.com/rsalevsky/barclamp-provisioner/blob/keep_hostname_improvments/chef/cookbooks/provisioner/templates/suse/crowbar_register.erb#L254

So you need to update the code there.